src/test/java/org/openhab/ui/habot/test/TrainerTest.java

```diff
@@ -135,5 +146,52 @@ public void testDE() throws Exception {
         assertEquals("activate-object", actual.getName());
         assertEquals(1, actual.getEntities().size());
         assertEquals("Licht", actual.getEntities().get("object"));
-    }    
+    }  
+
+    @Test
+    public void testDE_deactivateObjects() throws Exception {
+
+    	Intent actual;
+        this.trainer = new IntentTrainer("de", skills);
+
+        actual = interpret("Licht aus");
+        assertEquals("deactivate-object", actual.getName());
+        assertEquals(1, actual.getEntities().size());
+        assertEquals("Licht", actual.getEntities().get("object"));
+        
+        actual = interpret("mach den Fernseher aus");
+        assertEquals("deactivate-object", actual.getName());
+        assertEquals(1, actual.getEntities().size());
+        assertEquals("fernseher", actual.getEntities().get("object"));   
+        
+        actual = interpret("bitte mache das Licht aus");
+        assertEquals("deactivate-object", actual.getName());
+        assertEquals(1, actual.getEntities().size());
+        assertEquals("Licht", actual.getEntities().get("object"));
+    }  
+    
+    @Test
+    public void testDE_getStatus() throws Exception {
+
+    	Intent actual;
+        this.trainer = new IntentTrainer("de", skills);
+
+        actual = interpret("Heizung in der Küche");
+        assertEquals("get-status", actual.getName());
```

Comment:
I have extended the tests to validate the results.
In ```en``` and ```fr``` the entities are filled correct but in ```de``` not jet. Not sure if it is lac of training text or something else.

Reply:
Sometimes people can't be bothered to use proper caps while chatting, so I enforced the use of lowercase which now might be a problem with German nouns.
If I remove `.toLowerCase()` in 
https://github.com/ghys/habot/blob/0f121ce89ed962f825560f8e3bd31ede234d792a/src/main/java/org/openhab/ui/habot/nlp/internal/IntentTrainer.java#L140 and https://github.com/ghys/habot/blob/0f121ce89ed962f825560f8e3bd31ede234d792a/src/main/java/org/openhab/ui/habot/nlp/internal/IntentTrainer.java#L161 
your tests start working:

```
----
"Heizung in der Küche"
{0.7605032944838056=[get-status], 0.06591296782111584=[get-history-daily], 0.05018176282451643=[get-history-hourly, get-history-weekly], 0.025948107032804774=[deactivate-object, get-history-last-changes], 0.014565227252951047=[get-history-monthly], 0.006758770727485172=[activate-object]}
Intent [name=get-status, entities={location=Küche, object=Heizung}]
----
"wie hoch ist die Temperatur im Wohnzimmer"
{0.9459791986342562=[get-status], 0.03205956200147405=[activate-object], 0.007466023980015509=[get-history-last-changes], 0.00464473652974734=[get-history-hourly, get-history-weekly], 0.0024305142682756182=[deactivate-object], 0.0013876140282419193=[get-history-monthly, get-history-daily]}
Intent [name=get-status, entities={location=Wohnzimmer, object=Temperatur}]
----
"wie hoch ist die Temperatur im Keller"
{0.9459791986342562=[get-status], 0.03205956200147405=[activate-object], 0.007466023980015509=[get-history-last-changes], 0.00464473652974734=[get-history-hourly, get-history-weekly], 0.0024305142682756182=[deactivate-object], 0.0013876140282419193=[get-history-monthly, get-history-daily]}
Intent [name=get-status, entities={location=Keller, object=Temperatur}]
```


Maybe it makes sense to make everything lowercase in the German training data?

Reply:
Ah, okay it is case sensitiv! 
Hm, for future maintenance I would prefer to do the lower case within the system befor loading the data. Then it is save for future updates/languages, too.

Reply:
Makes sense.
The maybe best way to go would be to derive the `PlainTextByLineStream` class from OpenNLP and override its `read` method:
```
  @Override
  public String read() throws IOException {
    String line = in.readLine()
    return (line ? line.toLowerCase() : null);
  }
```
Then change the references of `PlainTextByLineStream` (twice) in `IntentTrainer` to the derived class ("`LowercasePlainTextByLineStream`" or similar).